### PR TITLE
SDI-152 Unit tests for all new service 

### PR DESCRIPTION
**Pull request by andymarke**
These complement the existing integration test but add more error scenarios

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderIepLevel.java [hunk 1]
```diff
@@ -33,12 +32,15 @@
 @Getter
 @RequiredArgsConstructor
 @AllArgsConstructor
-@Builder
 @Table(name = "OFFENDER_IEP_LEVELS")
 @Entity
 @ToString
 @IdClass(OffenderIepLevel.PK.class)
 public class OffenderIepLevel extends AuditableEntity {
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderIepLevel.java:39]: decided to de-lombok the builder - the only lombok feature Kotlin can not see

File: src/main/java/uk/gov/justice/hmpps/prison/service/transfer/ExternalMovementTransferService.kt [hunk 2]
```diff
@@ -62,7 +62,7 @@ class ExternalMovementTransferService(
 
   private fun getAdmissionMovementType(): Result<MovementType> =
     movementTypeRepository.findByIdOrNull(MovementType.ADM)?.let { success(it) } ?: failure(
-      BadRequestException.withMessage("No ${MovementType.ADM} movement type found")
+      EntityNotFoundException.withMessage("No ${MovementType.ADM} movement type found")
     )
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/hmpps/prison/service/transfer/ExternalMovementTransferService.kt:66]: small bug the tests found, wrong exception being thrown

File: src/main/java/uk/gov/justice/hmpps/prison/service/transfer/IEPTransferService.kt [hunk 3]
```diff
@@ -18,17 +18,18 @@ class IEPTransferService(
   staffUserAccountRepository = staffUserAccountRepository,
   authenticationFacade = authenticationFacade
 ) {
-  fun resetLevelForPrison(booking: OffenderBooking, transferMovement: ExternalMovement) {
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/hmpps/prison/service/transfer/IEPTransferService.kt:21]: small adjustment to return back the changed booking; easier to test that way

File: src/test/java/uk/gov/justice/hmpps/prison/service/transfer/BedAssignmentTransferServiceTest.kt [hunk 4]
```diff
@@ -19,24 +21,29 @@ internal class BedAssignmentTransferServiceTest {
   private val bedAssignmentHistoryRepository: BedAssignmentHistoriesRepository = mock()
   private val service = BedAssignmentTransferService(bedAssignmentHistoryRepository)
 
-  @Test
```
Comment (andymarke) [inline src/test/java/uk/gov/justice/hmpps/prison/service/transfer/BedAssignmentTransferServiceTest.kt:22]: just moved inside inner class

**Review by Mjwillis — APPROVED** [review]
Nice work!

**Pull request merged**